node/src/components/block_proposer.rs

```diff
@@ -197,251 +281,118 @@ impl BlockProposer {
-        self.state
+        self.sets
             .pending
             .retain(|deploy_hash, _| !deploys.contains_key(deploy_hash));
-        self.state.finalized_deploys.extend(deploys);
+        self.sets.finalized_deploys.extend(deploys);
     }
 
     /// Handles finalization of a block.
     fn handle_finalized_block<I, REv>(
         &mut self,
-        effect_builder: EffectBuilder<REv>,
+        _effect_builder: EffectBuilder<REv>,
         height: BlockHeight,
         deploys: I,
     ) -> Effects<Event>
     where
         I: IntoIterator<Item = DeployHash>,
-        REv: ReactorEventT,
     {
         self.finalized_deploys(deploys);
-        self.state.next_finalized = height + 1;
+        self.sets.next_finalized = height + 1;
 
-        if let Some(requests) = self.state.request_queue.remove(&self.state.next_finalized) {
+        if let Some(requests) = self.request_queue.remove(&self.sets.next_finalized) {
             requests
                 .into_iter()
-                .flat_map(|request| self.get_chainspec(effect_builder, request))
+                .flat_map(|request| {
+                    request
+                        .responder
+                        .respond(self.propose_deploys(
+                            self.deploy_config,
+                            request.current_instant,
+                            request.past_deploys,
+                        ))
+                        .ignore()
+                })
                 .collect()
         } else {
             Effects::new()
         }
     }
 
-    /// Prunes expired deploy information from the BlockProposer, returns the total deploys pruned
-    fn prune(&mut self, current_instant: Timestamp) -> usize {
-        self.state.prune(current_instant)
+    /// Checks if a deploy is valid (for inclusion into the next block).
+    fn is_deploy_valid(
+        &self,
+        deploy: &DeployHeader,
+        block_timestamp: Timestamp,
+        deploy_config: &DeployConfig,
+        past_deploys: &HashSet<DeployHash>,
+    ) -> bool {
+        let all_deps_resolved = || {
+            deploy.dependencies().iter().all(|dep| {
+                past_deploys.contains(dep) || self.sets.finalized_deploys.contains_key(dep)
+            })
+        };
+        let ttl_valid = deploy.ttl() <= deploy_config.max_ttl;
+        let timestamp_valid = deploy.timestamp() <= block_timestamp;
+        let deploy_valid = deploy.timestamp() + deploy.ttl() >= block_timestamp;
+        let num_deps_valid = deploy.dependencies().len() <= deploy_config.max_dependencies as usize;
+        ttl_valid && timestamp_valid && deploy_valid && num_deps_valid && all_deps_resolved()
     }
-}
-
-impl<REv> Component<REv> for BlockProposer
-where
-    REv: ReactorEventT,
-{
-    type Event = Event;
-    type ConstructionError = Infallible;
 
-    fn handle_event(
+    /// Returns a list of candidates for inclusion into a block.
+    // TODO:
+    /// rename to proposed deploys
```

Comment:
Is that TODO outdated?

Reply:
It was added by @dwerner , I just copied it verbatim.